integration/reuse_layer_rebuild_test.go

```diff
@@ -342,4 +342,158 @@ func testReusingLayerRebuild(t *testing.T, context spec.G, it spec.S) {
 			Expect(secondImage.Buildpacks[6].Layers["assets"].SHA).NotTo(Equal(firstImage.Buildpacks[6].Layers["assets"].SHA))
 		})
 	})
+
+	context("when an app is rebuilt and there is a change on the the extra assets", func() {
+		it("rebuilds the layer", func() {
+			var (
+				err         error
+				logs        fmt.Stringer
+				firstImage  occam.Image
+				secondImage occam.Image
+
+				firstContainer  occam.Container
+				secondContainer occam.Container
+			)
+
+			err = os.MkdirAll(filepath.Join(source, "custom-assets"), os.ModePerm)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = os.WriteFile(filepath.Join(source, "custom-assets", "hello.js"), []byte("console.log(\"Hello, world!\")\n"), 0755)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Same comment about moving test set up into an `it.Before`

Reply:
Similar to my comment in the unit tests, I think it would be worth setting up a new `context` for the custom assets and moving the test setup and tests into that context.

There's no "right" answer here, as we're trying to flatten a multi-dimensional testing space (i.e. the intersection of assertions like reusing layers with features like providing custom assets) into a single, linearized graph (i.e. nested contexts). Whichever way we do this will always feel a bit clunky, and like we're repeating ourselves.